### ISSUE TITLE
Implement the crossentropy loss function

### ISSUE THREAD

**Issue by finlaymorrison** [finlaymorrison/BunjiLearn#1]
Implement the crossentropy loss function as defined by [ml-cheatsheet](https://ml-cheatsheet.readthedocs.io/en/latest/loss_functions.html).
